src/python/tests/test_hal_gene_liftover.py

```diff
@@ -0,0 +1,191 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `hal_gene_liftover.py` script.
+
+The unit testing is divided into one test class for all top-level functions, and one test class per
+class found in this script; and within each test class, one test method per public function/method.
+
+Typical usage example::
+
+    $ pytest test_hal_gene_liftover.py
+
+"""
+import filecmp
+from importlib.abc import Loader
+from importlib.machinery import ModuleSpec
+from importlib.util import module_from_spec, spec_from_file_location
+from pathlib import Path
+import sys
+from types import ModuleType
+from typing import ContextManager, Dict, Union
+
+import pytest
+from pytest import raises
+
+if sys.version_info >= (3, 7):
+    from contextlib import nullcontext as does_not_raise
+else:
+    from contextlib import ExitStack as does_not_raise
+
+
+def import_module_from_file(module_file: Union[Path, str]) -> ModuleType:
+    """Import module from file path.
+
+    The name of the imported module is the basename of the specified module
+    file without its extension.
+
+    In addition to being returned by this function, the imported module is
+    loaded into the sys.modules dictionary, allowing for commands such as
+    :code:`from <module> import <class>`.
+
+    Args:
+        module_file: File path of module to import.
+
+    Returns:
+        The imported module.
+
+    """
+    if not isinstance(module_file, Path):
+        module_file = Path(module_file)
+    module_name = module_file.stem
+
+    module_spec = spec_from_file_location(module_name, module_file)
+
+    if not isinstance(module_spec, ModuleSpec):
+        raise ImportError(f"ModuleSpec not created for module file '{module_file}'")
+    if not isinstance(module_spec.loader, Loader):
+        raise ImportError(f"no loader found for module file '{module_file}'")
+
+    module = module_from_spec(module_spec)
+    sys.modules[module_name] = module
+    module_spec.loader.exec_module(module)
+
+    return module
+
+
+script_path = Path(__file__).parents[3] / 'scripts' / 'hal_alignment' / 'hal_gene_liftover.py'
+import_module_from_file(script_path)
+
+# pylint: disable=import-error,wrong-import-position
+
+import hal_gene_liftover  # type: ignore
+from hal_gene_liftover import SimpleRegion
+
+# pylint: enable=import-error,wrong-import-position
+
+
+class TestHalGeneLiftover:
+    """Tests script hal_gene_liftover.py"""
+
+    ref_file_dir = None  # type: Path
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self) -> None:
+        """Loads necessary fixtures and values as class attributes."""
+        type(self).ref_file_dir = (
+                    pytest.files_dir / 'scripts' / 'hal_alignment' / 'hal_gene_liftover')
```

Comment:
The `scripts` path component was to avoid a clash with a notional `ensembl.compara.hal_alignment` module,
but you are right, there is currently no need for it. 

I'm happy to leave it as just `hal_alignment`, on the assumption that other HAL-related scripts could benefit from
using the same test files.

The only tweak I might make would be to remove the parentheses, as these are no longer needed, since the path fits on 1 line.